crates/router/src/db/merchant_key_store.rs

```diff
@@ -95,18 +99,116 @@ impl MerchantKeyStoreInterface for Store {
 impl MerchantKeyStoreInterface for MockDb {
     async fn insert_merchant_key_store(
         &self,
-        _merchant_key_store: domain::MerchantKeyStore,
-        _key: &Secret<Vec<u8>>,
+        merchant_key_store: domain::MerchantKeyStore,
+        key: &Secret<Vec<u8>>,
     ) -> CustomResult<domain::MerchantKeyStore, errors::StorageError> {
-        // [#172]: Implement function for `MockDb`
-        Err(errors::StorageError::MockDbError.into())
+        let mut locked_merchant_key_store = self.merchant_key_store.lock().await;
+
+        if locked_merchant_key_store
+            .iter()
+            .any(|merchant_key| merchant_key.merchant_id == merchant_key_store.merchant_id)
+        {
+            Err(errors::StorageError::MockDbError)?;
+        }
+
+        let merchant_key = storage::MerchantKeyStore {
+            merchant_id: merchant_key_store.merchant_id,
+            key: merchant_key_store.key.into(),
+            created_at: merchant_key_store.created_at,
+        };
```

Comment:
I think we can use this function instead => https://github.com/juspay/hyperswitch/blob/1afc54837d5988eaf41f434474c30ec511681bbe/crates/router/src/types/domain/merchant_key_store.rs#L26
```suggestion
        let merchant_key: storage::MerchantKeyStore =
            Conversion::convert(merchant_key_store)
                .await
                .change_context(errors::StorageError::MockDbError)?;
```